src/screens/SignUp.js

```diff
@@ -0,0 +1,213 @@
+import React from "react";
+import { StyleSheet, View, Text, Image, ImageBackground } from "react-native";
+
+import LoginButton from "../components/LoginButton";
+import InputWithValidation from "../components/InputWithValidation";
+import firebase from "../config/fbConfig";
+import bgImage from "../images/background.jpg";
+import logo from "../images/logo.png";
+import {
+  validateEmail,
+  validatePhoneNumber,
+  validatePassword,
+} from "../helpers/validators";
+import ROUTES from "../consts/routes";
+import { ScrollView } from "react-native-gesture-handler";
+
+class SignUp extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: "",
+      username: "",
+      phoneNumber: "",
+      password: "",
+      errors: {},
+    };
+  }
+
+  handleValueChange = (name, value) => {
```

Comment:
5 стрічок, коли можна записати одною
```suggestion
  handleValueChange = (name, value) => this.setState({ [name]: value });
```